hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSender.java

```diff
@@ -89,18 +89,26 @@ public void shutdown() {
 
 
     public void sendAsync(Message message) {
-        sendAsync(message, 0);
+        sendAsync(message, delayForSubscription());
     }
 
     private void sendAsync(Message message, int delayMillis) {
         retrySingleThreadExecutor.schedule(() -> sendMessage(message), delayMillis, TimeUnit.MILLISECONDS);
     }
 
+    private int delayForSubscription() {
+        if (!subscription.isBatchSubscription()) {
```

Comment:
AFAIR `ConsumerMessageSender` is used only for serial subscriptions, so this if is redundant

Reply:
Yep, it's used only in `SerialConsumer`.